server/historian/packages/historian-base/src/services/restGitService.ts

```diff
@@ -237,8 +238,16 @@ export class RestGitService {
 			summaryParams,
 			initial !== undefined ? { initial } : undefined,
 		);
-		// JSON.stringify is not ideal for performance here, but it is better than crashing the cache service.
-		const summarySize = JSON.stringify(summaryResponse).length;
+
+		const summarySize = sizeof(summaryResponse);
```

Comment:
The sizeof package do followings based on https://www.npmjs.com/package/object-sizeof
> Node.js version uses the Buffer.from(objectToString) method to convert the object's string representation to a buffer, and then it uses the byteLength property to obtain the buffer size in bytes.

I think it will still do object sterilization, except before was doing entire object sterilization but using the package it is doing property by property.

What the use case for summarySize? do we care for it? I think http request logs should have the request size logged and we can correlate the size here for reporting?

Reply:
We are not using the summary size only for logging. We have to calculate it anyways to decide whether or not to cache the summary.

![image](https://github.com/microsoft/FluidFramework/assets/448106/aa3d3801-25a5-4359-a228-9acc48ba1ed0)

From my reading, it seems like when we are dealing with large objects, json stringify is a lot less efficient in terms of memory and it is also less accurate.